packages/block-library/src/gallery/gallery.js

```diff
@@ -104,4 +102,12 @@ export const Gallery = ( props ) => {
 	);
 };
 
+function RichTextVisibilityHelper( { isHidden, className, ...richTextProps } ) {
+	return isHidden ? (
+		<VisuallyHidden as={ RichText } { ...richTextProps } />
+	) : (
+		<RichText className={ className } { ...richTextProps } />
```

Comment:
I wanted to avoid passing className to `<VisuallyHidden />`, but when I looked into it more `.block-gallery-caption` shouldn't interfere with any styles we have there so I'll simplify this.